frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.jsx

```diff
@@ -153,6 +172,11 @@ export class SmartScalar extends Component {
     if (!insight) {
       return null;
     }
+    const { unit } = insight;
```

Comment:
I remember we discussed the visualization itself can infer this information from data. Why did we decide to go the other way?

Reply:
This is currently in progress for the next PR.